sklearn/model_selection/tests/test_search.py

```diff
@@ -1421,6 +1426,34 @@ def test_grid_search_cv_splits_consistency():
                        cv=KFold(n_splits=n_splits))
     gs2.fit(X, y)
 
+    assert_true(isinstance(KFold(n_splits=n_splits,
+                                 shuffle=True, random_state=0).split(X, y),
+                           GeneratorType))
+
+    # Give generator as a cv parameter
```

Comment:
There is a test a few lines above (at L1431), which confirms if KFold indeed returns a `GeneratorType`